examples/filter.py

```diff
@@ -0,0 +1,48 @@
+'''
+Takes in any data from a sensor and filters it.
+'''
+
+import json
+import zmq
+
+
+def bayes_filter(prev, measured):
+    ''' Super lame Bayes filter'''
+    return (prev + measured)/2
+
+
+def main():
+    # {{{ wrap this stuff
+    context = zmq.Context()
+    subscriber = context.socket(zmq.SUB)
+    subscriber.connect("tcp://localhost:5000")
+    subscriber.setsockopt(zmq.SUBSCRIBE, b"sensor")
+
+    kill_sig = context.socket(zmq.SUB)
+    kill_sig.connect("tcp://localhost:9000")
+    kill_sig.setsockopt(zmq.SUBSCRIBE, b"EXIT")
+    # }}}
+
+    prev = 1.5
+
+    try:
+        while True:
+            # {{{ wrap this stuff
+            try:  # Replace with polling?
```

Comment:
I know you can receive multiple different subscriptions from the same publisher, but I'm not sure about multiple publishers (many to one). Callbacks would definately be great.